src/shared_realm.cpp

```diff
@@ -185,12 +185,20 @@ void Realm::open_with_config(const Config& config,
         throw std::logic_error("Realms opened in Additive-only schema mode do not use a migration function");
     if (config.schema_mode == SchemaMode::ReadOnly && config.migration_function)
         throw std::logic_error("Realms opened in read-only mode do not use a migration function");
+    if (!config.realm_data.is_null() && !config.read_only())
```

Comment:
We should attempt to detect and reject other invalid config combinations here as well. For instance, setting both `path` and `realm_data`, or `realm_data` while providing an encryption key.